R/config.R

```diff
@@ -1,45 +1,126 @@
-check_yaml_has <- function(missing, where, pkg, call = caller_env()) {
-  if (length(missing) == 0) {
-    return()
-  }
+config_pluck <- function(pkg, path, default = NULL) {
+  check_string(path, allow_empty = FALSE, .internal = TRUE)
 
-  missing_components <- lapply(missing, function(x) c(where, x))
-  msg_flds <- purrr::map_chr(missing_components, paste, collapse = ".")
+  where <- strsplit(path, ".", fixed = TRUE)[[1]]
+  purrr::pluck(pkg$meta, !!!where, .default = default)
+}
 
-  config_abort(
-    pkg, 
-    "Can't find {cli::qty(missing)} component{?s} {.field {msg_flds}}.",
-    call = call
+config_pluck_character <- function(pkg,
```

Comment:
could this be called `config_pluck_chr()`

Reply:
and the string one, `config_pluck_str()`?

Reply:
I think these are more matches to rlang's `check_` helpers than purrr's, because they're about selecting single values (that are sometimes vectors).